third_party/move/tools/move-lint/src/lint/rules/bool_comparison.rs

```diff
@@ -0,0 +1,157 @@
+                        )),
+                        (Operation::Eq, false) | (Operation::Neq, true) => Some(format!(
+                            "Use !{} directly instead of comparing it to {}.",
+                            var_name, a
+                        )),
+                        _ => None,
+                    };
+
+                    if let Some(diagnostic_msg) = diagnostic_msg {
+                        add_diagnostic_and_emit(
+                            &env.get_node_loc(cond.node_id()),
+                            &diagnostic_msg,
+                            codespan_reporting::diagnostic::Severity::Warning,
+                            env,
+                            diags,
+                        );
+                    }
+                } else {
+                    for exp in args {
+                        self.check_boolean_comparison(exp, func_env, env, diags);
+                    }
+                }
+            }
+
+            if let Some(exp_val) = args.get(1) {
+                if let ExpData::Value(_, Value::Bool(b)) = &exp_val.as_ref() {
+                    let var_name = self
+                        .get_var_name_or_func_name_from_exp(&args[0], func_env, env)
+                        .expect("Expected to get a variable name");
+
+                    let diagnostic_msg = match (oper, b) {
+                        (Operation::Eq, true) | (Operation::Neq, false) => Some(format!(
+                            "Use {} directly instead of comparing it to {}.",
+                            var_name, b
+                        )),
+                        (Operation::Eq, false) | (Operation::Neq, true) => Some(format!(
+                            "Use !{} directly instead of comparing it to {}.",
+                            var_name, b
+                        )),
+                        _ => None,
+                    };
+
+                    if let Some(diagnostic_msg) = diagnostic_msg {
+                        add_diagnostic_and_emit(
+                            &env.get_node_loc(cond.node_id()),
+                            &diagnostic_msg,
+                            codespan_reporting::diagnostic::Severity::Warning,
+                            env,
+                            diags,
+                        );
+                    }
+                } else {
+                    for exp in args {
+                        self.check_boolean_comparison(exp, func_env, env, diags);
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl ExpressionAnalysisVisitor for BoolComparisonVisitor {
+    fn post_visit_expression(
+        &mut self,
+        exp: &ExpData,
+        func_env: &FunctionEnv,
+        env: &GlobalEnv,
+        _: &LintConfig,
+        diags: &mut Vec<Diagnostic<FileId>>,
+    ) {
+        if let ExpData::IfElse(_, cond, _, _) = exp {
+            self.check_boolean_comparison(cond.as_ref(), func_env, env, diags);
```

Comment:
Boolean comparisons can also appear elsewhere, like `let y = (x == true);` or `foo(x == true)`. We can report the same warning in all these cases, I don't see any reason to limit to just conditionals.